src/renderer/shared/api/xcm/lib/xcm-utils.ts

```diff
@@ -152,48 +155,50 @@ function getConcreteAssetLocation(assetLocation?: LocalMultiLocation) {
 }
 
 function getDestinationLocation(
-  originChain: Pick<Chain, 'parentId'>,
+  originChain: Pick<Chain, 'parentId' | 'specName'>,
   destinationParaId?: number,
   accountId?: AccountId,
 ) {
   if (originChain.parentId && destinationParaId) {
-    return getSiblingLocation(destinationParaId, accountId);
+    return getSiblingLocation(destinationParaId, accountId, originChain.specName as Network);
   }
 
   if (originChain.parentId) {
-    return getParentLocation(accountId);
+    return getParentLocation(accountId, originChain.specName as Network);
   }
 
   if (destinationParaId) {
-    return getChildLocation(destinationParaId, accountId);
+    return getChildLocation(destinationParaId, accountId, originChain.specName as Network);
   }
 
   return undefined;
 }
 
-function getAccountLocation(accountId?: AccountId) {
+function getAccountLocation(accountId?: AccountId, network?: string) {
   const isEthereum = isEthereumAccountId(accountId);
 
   return {
     parents: 0,
     interior: {
-      X1: {
-        [isEthereum ? 'accountKey20' : 'accountId32']: {
-          network: 'Any',
-          [isEthereum ? 'key' : 'id']: accountId,
+      X1: [
+        {
+          [isEthereum ? 'accountKey20' : 'accountId32']: {
+            network: network || 'Any',
```

Comment:
Why do we choose specName to specify network here? looks like that parameteris is used for another pupose in XcmV3JunctionNetworkId, most of XCMs set here Null